scripts/processFactory.ts

```diff
@@ -0,0 +1,525 @@
+﻿/// <reference path="../src/compiler/sys.ts" />
+module ts {
+    interface Map<T> {
+        [name: string]: T;
+    }
+
+    interface SyntaxNode {
+        name?: string;
+        kind?: string;
+        type?: string;
+        types?: string;
```

Comment:
In the factory.json file I write the union types as “A | B”. It allowed me to reuse some other functionality in the script and keep the json somewhat more consistent.

From: Daniel Rosenwasser [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:37 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In scripts/processFactory.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24627729>:

> @@ -0,0 +1,525 @@

> +﻿/// <reference path="../src/compiler/sys.ts" />

> +module ts {

> +    interface Map<T> {

> +        [name: string]: T;

> +    }

> +

> +    interface SyntaxNode {

> +        name?: string;

> +        kind?: string;

> +        type?: string;

> +        types?: string;

types is a string and not astring[]`?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24627729>.